presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketing.java

```diff
@@ -300,6 +301,15 @@ private static int hashBytes(int initialValue, Slice bytes)
             return Optional.of(new HiveBucketFilter(ImmutableSet.of(singleBucket.getAsInt())));
         }
 
+        Optional<Map<ColumnHandle, Set<NullableValue>>> columnValueSetBindings = TupleDomain.extractFixedValuesFromRanges(effectivePredicate);
```

Comment:
@gparai It seems like the logic in lines 295-302 would be covered by a new more generic logic, no? E.g. that logic processing the case where the filter matches exactly one bucket, but the new logic allows for multiple buckets. Any reason to keep both?

Reply:
The new logic applies to SortedRangeSet. I was not sure if for the single bucket case we could end up with predicates having EquatableValueSet/AllOrNoneValueSet hence I had left it alone. For the single bucket case, it looks like we generate the SortedRangeSet so keeping only the new logic should be fine.